pymc_marketing/clv/utils.py

```diff
@@ -597,3 +598,185 @@ def rfm_train_test_split(
+            "Repeat Customer": Customers in top 2 quartiles for frequency, and either recency or monetary value.
+            "Top Spender": Customers in top 2 quartiles for monetary value, and either frequency or recency.
+            "At-Risk Customer": Customers in bottom 2 quartiles for two or more variables.
+            "Inactive Customer": Customers in bottom quartile for two or more variables.
+        Customers with unspecified RFM scores will be assigned to a segment named "Other".
+    observation_period_end: Union[str, pd.Period, datetime], optional
+        A string or datetime to denote the final date of the study.
+        Events after this date are truncated. If not given, defaults to the max 'datetime_col'.
+    datetime_format: string, optional
+        A string that represents the timestamp format. Useful if Pandas can't understand
+        the provided format.
+    time_unit: string, optional
+        Time granularity for study.
+        Default: 'D' for days. Possible values listed here:
+        https://numpy.org/devdocs/reference/arrays.datetime.html#datetime-units
+    time_scaler: int, optional
+        Default: 1. Useful for scaling recency & T to a different time granularity. Example:
+        With freq='D' and freq_multiplier=1, we get recency=591 and T=632
+        With freq='h' and freq_multiplier=24, we get recency=590.125 and T=631.375
+        This is useful if predictions in a different time granularity are desired,
+        and can also help with model convergence for study periods of many years.
+    sort_transactions: bool, optional
+        Default: True
+        If raw data is already sorted in chronological order, set to `False` to improve computational efficiency.
+
+    Returns
+    -------
+    :obj: DataFrame:
+        customer_id, frequency, recency, T [, monetary_value]
+    """
+
+    rfm_data = rfm_summary(
+        transactions,
+        customer_id_col=customer_id_col,
+        datetime_col=datetime_col,
+        monetary_value_col=monetary_value_col,
+        observation_period_end=observation_period_end,
+        datetime_format=datetime_format,
+        time_unit=time_unit,
+        time_scaler=time_scaler,
+        include_first_transaction=True,
+        sort_transactions=sort_transactions,
+    )
+
+    # change recency to segmentation definition
+    rfm_data["recency"] = rfm_data["T"] - rfm_data["recency"]
+
+    for column_name in zip(
+        ["r_quartile", "f_quartile", "m_quartile"],
+        ["recency", "frequency", "monetary_value"],
+        strict=False,
+    ):
+
+        def _quartile_labels(column_name, max_label_range):
+            # recency labels must be reversed because lower values are more desirable
+            if column_name == "r_quartile":
+                return list(reversed(list(range(1, max_label_range))))
+            else:
+                return range(1, max_label_range)
+
+        try:
+            labels = _quartile_labels(column_name[0], 4)
```

Comment:
Will there be any case where we would like to have `max_label_range != 4`?

Reply:
(and `max_label_range != 3`, `max_label_range != 5` below?

Reply:
Thank you for the explanation! It might be worth adding this comments in the doc-strings (or code comment)